scrapper.py

```diff
@@ -25,39 +36,159 @@ class Crawler:
+                              'Источники']
+
+        class_of_link = article_bs.find('div', {'class': "articles-item-more"})
+        if class_of_link:
+            link_to_save = class_of_link.find('a')
+            link_to_pdf = 'https://periodical.pstgu.ru' + link_to_save['href']
+            final_page = requests.get(link_to_pdf)
+            final_page_bs = BeautifulSoup(final_page.text, 'html.parser')
+        else:
+            final_page_bs = article_bs
+        url_pattern = re.compile(r"var DEFAULT_URL = '(.*?)'")
+        final_link = final_page_bs.find(string=url_pattern)
+        final_link = 'https://periodical.pstgu.ru' + \
+                     str(final_link).replace('var DEFAULT_URL = ', '').replace("'", '').replace(';', '').strip()
+        raw_pdf = PDFRawFile(final_link, self.article_id)
+        raw_pdf.download()
+        text_from_pdf = raw_pdf.get_text()
+        for refs_maker in list_of_possible_l:
+            if refs_maker in text_from_pdf:
+                text_pdf_list = text_from_pdf.split(refs_maker)
+                del text_pdf_list[-1]
+                text_from_pdf = ' '.join(text_pdf_list)
+        self.article.text = text_from_pdf
+
+    def _fill_article_with_meta_information(self, article_bs):
+        issue_number = []
+        month_dict = {2: '03', 3: '06', 0: '09', 1: '12'}
+        try:
+            author = article_bs.find('p', {"class": "articles-item-author"}).find('b')
+            self.article.author = author.text
+        except AttributeError:
+            self.article.author = 'No data'
+        title = article_bs.find('h2', {"class": "articles-item-name"})
+        self.article.title = title.text
+        topics = article_bs.find('div', {"class": "articles"}).find_all('p')
+        try:
+            self.article.topics = topics[2].text
+        except IndexError:
+            self.article.topics = 'No data'
+        year = article_bs.find('meta', {'name': 'citation_publication_date'})['content']
+        issue_title = article_bs.find('meta', {'name': 'citation_journal_title'})['content']
+        for letter in issue_title:
+            if letter.isdigit():
+                issue_number.append(letter)
+        month = month_dict[int(''.join(issue_number)) % 4]
+        text_date = month + '.' + year
+        date = datetime.datetime.strptime(text_date, '%m.%Y')
+        self.article.date = date
 
 
 def prepare_environment(base_path):
     """
     Creates ASSETS_PATH folder if not created and removes existing folder
     """
-    pass
+    base_path_p = pathlib.Path(base_path)
+    if os.path.exists(base_path_p):
+        shutil.rmtree(base_path_p)
+        base_path_p.mkdir(parents=True)
+    else:
+        base_path_p.mkdir(parents=True)
 
 
 def validate_config(crawler_path):
     """
     Validates given config
     """
-    pass
+    with open(crawler_path) as path_thing_idk:
+        config_dict = json.load(path_thing_idk)
+    seed_urls = config_dict['seed_urls']
```

Comment:
are you sure those keys will be always present in such a config? will you accept an incomplete config?